src/Interpreters/convertFieldToType.cpp

```diff
@@ -487,6 +488,18 @@ Field convertFieldToTypeImpl(const Field & src, const IDataType & type, const ID
             return object;
         }
     }
+    else if (const DataTypeVariant * type_variant = typeid_cast<const DataTypeVariant *>(&type))
+    {
+        /// If we have type hint and Variant contains such type, no need to convert field.
+        if (from_type_hint && type_variant->tryGetVariantDiscriminator(*from_type_hint))
+            return src;
+
+        /// Create temporary column and check if we can insert this field to the variant.
+        /// If we can insert, no need to convert anything.
+        auto col = type_variant->createColumn();
+        if (col->tryInsert(src))
```

Comment:
Is it the only usage of `tryInsert`? If it's used only here, can't we introduce more specific method that checks type of the Field, but do not insert

Reply:
Main usage of `tryInsert ` is `ColumnVariant::insert`, when we cannot determine to which variant we need to insert the field, so we just try to insert it to all variants in order